### PR TITLE
ROX-24700: Resolve CentOS 8 deprecation

### PR DESCRIPTION
**Pull request by BradLugo**
## Description

Dockerfiles/Containerfiles using `quay.io/centos/centos:stream8` as the base image were breaking due to the removal of `mirrorlist.centos.org`. 
Example: https://github.com/stackrox/rox-ci-image/actions/runs/10011868762/job/27676182918

These changes substitute `ubi8:latest` in place of `centos:stream8` and add another image, `stackrox-ui-test`, to enable X11 support, which is required for running UI tests (particularly in OSCI).

### Testing

Changes have been validated here: 
- https://github.com/stackrox/stackrox/pull/11491
- https://github.com/openshift/release/pull/54854
- https://github.com/stackrox/scanner/pull/1589
- https://github.com/stackrox/stackrox/pull/11489

#### CentOS Vault mirror

This is a valid option and probably the most reasonable way to get this working. If I had known about it when I first started this work, I would probably have done this just to get everything working again. But the longer-term fix is already in this PR.

#### Use UBI8 in the UI tests

Red Hat folks have explicitly stated that X11 packages will not be supported in UBI registries.

#### CentOS Stream 9 across the board

This is problematic due to `CGO` and `glibc` versions mismatching in CI. See the following PRs for more details:
- https://github.com/stackrox/rox-ci-image/pull/206
- https://github.com/stackrox/rox-ci-image/pull/209
- https://github.com/stackrox/stackrox/pull/12028
- https://github.com/openshift/release/pull/54432

### Future improvements

For a cleaner solution in the future, I propose we:

- Consolidate `stackrox-build` and `stackrox-test` into a single image that contains all tools that are required for development and running everything expected to work in GitHub Actions and Konflux
- Figure out all the necessary dependencies and packages for the above ^ remove everything not required
- Continue to use `stackrox-ui-test`, possibly remain to something more meaningful (maybe `stackrox-osci` or `stackrox-e2e-test`?), and also remove all unnecessary dependencies and packages after investigating them

### CODE REVIEW

**Comment by BradLugo** [comment]
A side note: it's not clear to me why we have `stackrox-build` and `stackrox-test`. I suspect there's a historical reason (here's the PR `stackrox-test` was introduced: https://github.com/stackrox/rox-ci-image/pull/112) to be more consistent with other OCP teams. But we use GitHub Actions in https://github.com/stackrox/stackrox to build our container images now, so I don't currently see a reason to keep both of them around.

**Comment by BradLugo** [comment]
Most of the testing has been done here:
- https://github.com/stackrox/stackrox/pull/11491
- https://github.com/openshift/release/pull/54854

File: images/stackrox-ui-test.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+# for local development style & lint.
+ARG PYCODESTYLE_VERSION=2.10.0
+ARG PYLINT_VERSION=2.13.9
+RUN set -ex \
+  && pip3 install pycodestyle=="${PYCODESTYLE_VERSION}" \
+                  pylint=="${PYLINT_VERSION}"
+
+RUN \
+	mv /bin/bash /bin/real-bash && \
+	mv /bin/bash-wrapper /bin/bash
```
Comment (RTann) [inline images/stackrox-ui-test.Dockerfile:218]: nit: newline

File: images/scanner-test.Dockerfile [hunk 2]
```diff
@@ -49,7 +48,6 @@ RUN dnf install -y \
         lsof \
         lz4 \
         openssl \
-        postgresql${PG_MAJOR}-server \
```
Comment (RTann) [inline images/scanner-test.Dockerfile:52]: why is this removed? It seems like postgres is removed from the image, and it's needed for DB testing

Comment (BradLugo) [reply images/scanner-test.Dockerfile:52]: I was testing something. Good catch

File: images/collector.Dockerfile [hunk 3]
```diff
@@ -25,7 +46,6 @@ RUN dnf update -y && \
         docker-ce \
         docker-ce-cli \
         docker-ce-rootless-extras \
-        docker-scan-plugin \
```
[the remark this answers is not in the document]
Comment (BradLugo) [reply images/collector.Dockerfile:28]: I couldn't find any usage in https://github.com/stackrox/collector, but let's confirm with @stackrox/collector-team

File: images/scanner-test.Dockerfile [hunk 4]
```diff
@@ -49,15 +49,14 @@ RUN dnf install -y \
...
         procps-ng \
         python3 \
         unzip \
         xz \
         zip \
-        && \
-    dnf clean all && \
-    rm -rf /var/cache/dnf /var/cache/yum
+        postgresql${PG_MAJOR}-server \
```
[the remark this answers is not in the document]
Comment (RTann) [reply images/scanner-test.Dockerfile:57]: nit: looks like this was originally in alphabetical order. Want to move it back?

File: Makefile [hunk 5]
```diff
@@ -11,8 +8,8 @@ STACKROX_BUILD_TAG=$(shell scripts/get_tag.sh "stackrox-build")
 .PHONY: stackrox-build-image
 stackrox-build-image:
 	$(DOCKER) build \
+		--platform linux/amd64 \
```
[the remark this answers is not in the document]
Comment (BradLugo) [reply Makefile:11]: Among other things, due to `https://download.postgresql.org/pub/repos/yum/reporpms/EL-8-x86_64/pgdg-redhat-repo-latest.noarch.rpm`, we only support `x86_64` (`EL-8-x86_64` in the URL). Without this setting, the make targets break on other archs (e.g., my m1 macbook 🙃)

File: images/scanner-test.Dockerfile [hunk 6]
```diff
@@ -34,10 +34,10 @@ ENV BASH_ENV /etc/initial-bash.env
 ENV PG_MAJOR=15
 ENV PATH="$PATH:/usr/pgsql-$PG_MAJOR/bin/"
 
-RUN dnf install -y \
-        https://download.postgresql.org/pub/repos/yum/reporpms/EL-8-x86_64/pgdg-redhat-repo-latest.noarch.rpm \
- && dnf -qy module disable postgresql \
- && dnf update -y \
+# Install Postgres repo
+RUN dnf --disablerepo="*" install -y https://download.postgresql.org/pub/repos/yum/reporpms/EL-9-x86_64/pgdg-redhat-repo-latest.noarch.rpm
```
[the remark this answers is not in the document]
Comment (BradLugo) [reply images/scanner-test.Dockerfile:38]: Could be. I'm pretty sure Postgres 15 is in both package repos. I figured we could move it right now, but I'm down to keep it at 8.

[the remark this answers is not in the document]
Comment (RTann) [reply images/scanner-test.Dockerfile:38]: let's keep it at 8 just to be on the safe side, so everything matches

File: images/collector.Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (BradLugo) [reply images/collector.Dockerfile:28]: Synced with the collector team. Doesn't seem to be needed. As you said, David, we can figure out a solution later if it ends up causing any problems.

**Review by RTann — APPROVED** [review]
Scanner images LGTM, thanks

**Review by tommartensen — APPROVED** [review]
